c++/mpi/mpi.hpp

```diff
@@ -41,7 +41,7 @@ namespace mpi {
    * as cray uses MPICH under the hood it should work as well
    */
   static const bool has_env = []() {
-    if (std::getenv("OMPI_COMM_WORLD_RANK") != nullptr or std::getenv("PMI_RANK") != nullptr)
+    if (std::getenv("OMPI_COMM_WORLD_RANK") != nullptr or std::getenv("pmi_rank") != nullptr or std::getenv("CRAY_MPICH_VERSION") != nullptr)
```

Comment:
Why `PMI_RANK` -> `pmi_rank` ?

Reply:
I am sorry, my mistake. Will push a reverting fix.

Reply:
There now it is reverted to `PMI_RANK`.